lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1178,7 +1178,10 @@ class HSAQueue final : public KalmarQueue
 					if (v != 0) {
 						return false;
 					}
-				}
+                } else {
+                    // no signal, have to assume the command is still running
+                    return false;
```

Comment:
Is this fix mostly for efficiency rather than for correctness?
Normally I'd think that eventually it will encounter a command with a signal?  I guess I'm not sure in which situation that the last command would not have a signal?

 I found one usage of isEmpty() in this file, which is in createOrstealRocrQueue.  Note there's a lock_guard before the isEmpty() call; however, there's no lock around pushAsyncOp() so other threads may call this to add new commands to the queue so this could be a race condition?

Also, the usage of lock_guard probably has a bug because it doesn't have a variable name so it would get destroyed right after it's constructed.